cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -257,20 +248,77 @@ private String getSchemaCompatibility(
     }
   }
 
+  private Pair<String, String> getSubjectCompatibilityWithGlobalFallBack(
+      String environmentVal,
+      String topicName,
+      KafkaSupportedProtocol protocol,
+      String clusterIdentification) {
+    try {
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String compatibility =
+          getSubjectSchemaCompatibility(environmentVal, topicName, protocol, clusterIdentification);
+
+      if (!compatibility.equals(SCHEMA_COMPATIBILITY_NOT_SET)) {
+        return Pair.of(SUBJECT, compatibility);
+      }
+
+      String suffixUrl = environmentVal + "/config/";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol);
+
+      ResponseEntity<Map<String, String>> responseList =
+          getSubjectSchemaCompatibility(
+              reqDetails, new HashMap<>(), createSchmeaRegistryRequest(clusterIdentification));
+      log.info("Schema compatibility " + responseList);
+      return Pair.of(GLOBAL, responseList.getBody().get("compatibilityLevel"));
+    } catch (Exception e) {
+      log.error("Error in getting schema compatibility ", e);
+      return Pair.of(SCHEMA_COMPATIBILITY_NOT_SET, null);
+    }
+  }
+
+  private HttpEntity<Object> createSchmeaRegistryRequest(String clusterIdentification) {
+    HttpHeaders headers =
+        clusterApiUtils.createHeaders(clusterIdentification, KafkaClustersType.SCHEMA_REGISTRY);
+    HttpEntity<Object> request = new HttpEntity<>(headers);
+    return request;
+  }
+
+  private static ResponseEntity<Map<String, String>> getSubjectSchemaCompatibility(
+      Pair<String, RestTemplate> reqDetails,
+      Map<String, String> params,
+      HttpEntity<Object> request) {
+    return reqDetails
+        .getRight()
+        .exchange(
+            reqDetails.getLeft(),
+            HttpMethod.GET,
+            request,
+            GET_SCHEMA_COMPATIBILITY_TYPEREF,
```

Comment:
```suggestion
            PARAMETERIZED_TYPEREF,
```

Looks like a generic one, and not specific to schema

Reply:
This one is specific just to schema.